src/export/bops/index.ts

```diff
@@ -0,0 +1,536 @@
+  applicant_first_name: "applicant.name.first",
+  applicant_last_name: "applicant.name.last",
+  applicant_phone: "applicant.phone.primary",
+  applicant_email: "applicant.email",
+
+  agent_first_name: "applicant.agent.name.first",
+  agent_last_name: "applicant.agent.name.last",
+  agent_phone: "applicant.agent.phone.primary",
+  agent_email: "applicant.agent.email",
+
+  description: "proposal.description",
+};
+
+function isTypeForBopsPayload(type?: ComponentType) {
+  if (!type) return false;
+
+  switch (type) {
+    case ComponentType.Calculate:
+    case ComponentType.Confirmation:
+    case ComponentType.Content:
+    case ComponentType.DrawBoundary:
+    case ComponentType.ExternalPortal:
+    case ComponentType.FileUpload:
+    case ComponentType.MultipleFileUpload:
+    case ComponentType.Filter:
+    case ComponentType.FindProperty:
+    case ComponentType.Flow:
+    case ComponentType.InternalPortal:
+    case ComponentType.Notice:
+    case ComponentType.Pay:
+    case ComponentType.PlanningConstraints:
+    case ComponentType.PropertyInformation:
+    case ComponentType.Answer:
+    case ComponentType.Result:
+    case ComponentType.Review:
+    case ComponentType.Section:
+    case ComponentType.Send:
+    case ComponentType.SetValue:
+    case ComponentType.TaskList:
+      return false;
+
+    case ComponentType.AddressInput:
+    case ComponentType.Checklist:
+    case ComponentType.DateInput:
+    case ComponentType.NumberInput:
+    case ComponentType.Question:
+    case ComponentType.TextInput:
+    case ComponentType.ContactInput:
+      return true;
+
+    default:
+      return exhaustiveCheck(type);
+  }
+}
+
+function exhaustiveCheck(type: never): never {
+  throw new Error(`Unhandled type ${type}`);
+}
+
+// For a given node (a "Question"), recursively scan the flow schema to find which portal it belongs to
+//   and add the portal_name to the QuestionMetadata so BOPS can group proposal_details
+const addPortalName = (
+  id: string,
+  flow: LooseFlowGraph,
+  metadata: QuestionMetaData
+): QuestionMetaData => {
+  if (id === "_root") {
+    metadata.portal_name = "_root";
+  } else if (flow[id]?.type === ComponentType.InternalPortal) {
+    // internal & external portals are both type 300 after flattening (ref dataMergedHotFix)
+    metadata.portal_name = flow[id]?.data?.text || id;
+  } else {
+    // if the current node id is not the root or a portal, then find its' next parent node and so on until we hit a portal
+    Object.entries(flow).forEach(([nodeId, node]) => {
+      if (node.edges?.includes(id)) {
+        addPortalName(nodeId, flow, metadata);
+      }
+    });
+  }
+
+  return metadata;
+};
+
+// const addSectionName = (
+//   id: string,
+//   metadata: QuestionMetaData
+// ): QuestionMetaData => {
+//   const { hasSections, getSectionForNode } = useStore.getState();
+//   if (hasSections) {
+//     metadata.section_name = getSectionForNode(id).data.title;
+//   }
+//   return metadata;
+// };
```

Comment:
This is a known TODO & will require bringing over a few methods from the `NavigationStore`. Since sections are not actually in use yet on LDCs and therefore not actually being sent to BOPS, I've pushed this off. I can tack it onto these changes, or pick up separately in a small followup PR.

Reply:
Thanks for the clarification here - a follow on PR makes perfect sense to me. 

This ticket is marked as blocked until we resolve this - https://trello.com/c/Cb5nFHIi/2297-99-share-section-names-with-bops